proposals/20230605-rules-adoption-management-maturity-framework.md

```diff
@@ -0,0 +1,222 @@
+
+Newcomers to Falco will be encouraged to start by configuring their setup with introductory rules labeled as "Falco's default rules" (maturity: "Stable"). These rules will be based on syscall and container events and will live in the established [falco_rules.yaml](https://github.com/falcosecurity/rules/blob/main/rules/falco_rules.yaml). 
+
+As users become more familiar with Falco and better understand their unique environments, they can gradually fine-tune the default rules to meet their specific requirements. Tuning rules goes hand in hand with assessing the performance overhead and adjusting Falco's [configuration](https://github.com/falcosecurity/falco/blob/master/falco.yaml) accordingly. This consideration is crucial to convey to adopters, as it is important to keep in mind that there are usually limitations to the budget allocated for security monitoring.
+
+Once adopters have integrated the stable default rules with low false positives and acceptable performance consistently, they can add a next set of rules (maturity: "Incubating") for more specific detections. These efforts will be accompanied by guidance on rule customization and broader educational initiatives focusing on Linux OS runtime security.
+
+> Experts in the field, with expertise in areas such as offensive security, Linux kernel tracing, production deployment, cyber security, threat detection, compliance, data analysis, and data science, will be responsible for assessing the maturity level of rules. These experts have a deep familiarity with Falco. 
+
+
+### Define Rule Contribution Process
+
+A guide will be established to facilitate the contribution of new rules or updates to existing ones. This guide will serve as an extension to the generic Falco [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md). This guide will outline clear acceptance criteria for each maturity level and establish minimum requirements for rule creation. These criteria will be utilized to assess the acceptance or rejection of a rule.
+
+This guide will also include information on how the community is intended to share rules. As a general principle, the project's main objective is to curate an optimal set of "Stable" rules that are included in the main [falco_rules.yaml](https://github.com/falcosecurity/rules/blob/main/rules/falco_rules.yaml) file and also published as OCI artifacts. However, there may be opportunities to create additional platforms for ad-hoc rules sharing that do not necessarily need to meet the strict criteria set by the Falco maintainers.
+
+
+The minimum criteria for rules creation shall include:
+
+*Correctness*
+
+The rule must be both syntactically and grammatically correct and should evaluate to true during successful end-to-end tests. Furthermore, it needs to accurately detect the intended cyber threats, specifically the Tactics, Techniques, and Procedures (TTPs).
```

Comment:
@incertum I am writting down some things here that we discussed about Mitre to trace it.
So beyond the correctness of the rules regarding the proper Falco syntax, I would propose more precisions about the correctness of Falco rules regarding standards (or de facto standards), like the Mitre ATT&CK, when the rules are using extra tags.

The rules already use the naming system `mitre_<mitre phase name>`, so maybe we can stick with it to detect when a rule is using a Mitre extra tag, or simply a technique id like `T1021.004`.

I am still pretty confident with STIX2 standard to fetch the Mitre ATT&CK data, so maybe it can appear in this section ?
Something like :

"The rule must be both syntactically and grammatically correct and should evaluate to true during successful end-to-end tests. 
Furthermore, it needs to accurately detect the intended cyber threats, specifically the Tactics, Techniques, and Procedures (TTPs), _against the Mitre ATT&CK framework.
Both the community and the Falco experts will benefit from the falco mitre checker module to audit the default rules or custom rules against the STIX2 data from Mitre CTI._ "

Reply:
Thank you a lot @IceManGreen I'll work it in and leave it more general for now and then we decide on the technical details for the Falco Mitre checker module you are looking to create in your proposal ❤️ !